File: Practica1/Ejercicio2/FactoriaMontana.py
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from CarreraMontana import CarreraMontana
from BicicletaMontana import BicicletaMontana

class FactoriaMontana(FactoriaCarreraYBicicleta):
    def crear_carrera(self, nombre, bicicletas):
        carrera_de_montana = CarreraMontana(nombre, bicicletas)
        return carrera_de_montana
    
    def crear_bicicleta(self, id):
        bicicleta_de_montana = BicicletaMontana(id)
        return bicicleta_de_montana

File: Practica1/Ejercicio1/src/FactoriaMontana.java
import java.util.ArrayList;

public class FactoriaMontana implements FactoriaCarreraYBicicleta {
    public FactoriaMontana() {

    }

    public CarreraMontana crearCarrera(String nombre, ArrayList<Bicicleta> bicicletas) {
        CarreraMontana objCarreraMontana = new CarreraMontana(nombre, bicicletas);

        return objCarreraMontana;
    }

    public BicicletaMontana crearBicicleta(int id) {
        BicicletaMontana objBicicletaMontana = new BicicletaMontana(id);

        return objBicicletaMontana;
    }

}